Configure 'test-bare' script

medikoo opened the pull request:
The problem with `test` script is that it involves coverage wrap and eventual error reports do not map to code locations as in files.

In projects I maintain I usually keep following convention:
- `test` - bare tests
- `coverage` - coverage report (obviously comes with test run, and fails if tests fail, so equivalent to `test` as it's now)
- `check-coverage` - Runs `coverage` and fails if coverage is below predefined threshold

Then in CI config, I configure one run to use `check-coverage` and others to run `test` (e.g. when testing multiple Node.js version). If you like it I can also update it to reflect that.

dougmoscrop commented:
Yeah, ostensibly there's source maps that are supposed to help but they don't seem to work or I've not configured them.

I'm fine with making test just call ava, and adding a second ci script/command that does the nyc part.

medikoo commented:
@dougmoscrop I've updated it so there are now three tasks configured independently: `test`, `lint`, `coverage`.

Additionally I've updated CI configuration to run `lint` and `test` independently (and in parallel). Additionally it's nice, as those two steps, are now shown as two independent checks in PR status :)

Let me know if that's fine

dougmoscrop commented:
Thanks!
